### ISSUE TITLE
KameletBinding - Make error handler more simpler

### ISSUE THREAD

**Issue by davsclaus** [apache/camel-k#2815]
A binding is intended to be high level and fairly simple.
https://camel.apache.org/camel-k/1.7.x/kamelets/kameletbindings-error-handler.html

I wonder if we have not made the error handler a bit too complex with the `bean` and `ref`. What is the purpose of those. As they require the user to implement a custom error handler builder (which nobody really does in core camel).

I wonder if we can deprecate and remove those, or did you find a use-case for them somehow in camel-k?

**Comment by davsclaus** [comment]
@squakez any thoughts?

**Comment by squakez** [comment]
Yeah, I think the direct usage of them can be hidden to the final user, but cannot be removed (at least `ref`) as it's the foundation of the feature itself (see https://github.com/apache/camel-k/blob/2ef783640db6c2cb57661cf944fc67360e767dc0/pkg/trait/error_handler.go#L97). When I developed originally the feature, I exposed all the low level implementations to collect feedback on possible advanced use case. I don't see any added value to the user, so, I agree we can remove the option. Also @lburgazzoli was involved on the design, maybe he had in mind some other use case I ignore.
